### PR TITLE
TimeHandler

### PR DESCRIPTION
**Pull request by meteguerlek**
again

### CODE REVIEW

**Comment by fletch3555** [comment]
Still not on a feature branch, but at least this one can be merged (once reviewed).  Please avoid committing anything to your `1.0` branch until this gets merged

**Comment by meteguerlek** [comment]
Ok sorry, thanks

**Review by marktopper — APPROVED** [review]

**Pull request merged**
